crates/models/src/build/mod.rs

```diff
@@ -27,16 +28,289 @@ pub fn inference(shape: &Shape, exists: Exists) -> flow::Inference {
+
+    // Until there's a good reason otherwise, we hard-code that fragments are refreshed every five minutes.
+    let refresh_interval = Some(Duration::from_secs(5 * 60).into());
+
+    // If an explicit retention interval isn't provided, then don't set one.
+    let retention = retention.map(Into::into);
+
+    // Partition journals are readable and writable.
+    // We could get fancier here by disabling writes to a journal which has no captures.
+    let flags = broker::journal_spec::Flag::ORdwr as u32;
+
+    // We hard-code max_append_rate to 4MB/s, which back-pressures derivations
+    // that produce lots of register updates. They'll hopefully perform more
+    // aggregation per-transaction, and eventually stall until there's quota.
+    // TODO(johnny): I sized this to be over the steady-state of our current
+    // use cases. We'll want to revisit this value when we have better tooling
+    // for splitting journals, and are able to study its cumulative effects
+    // with varietes of journals and use cases.
+    let max_append_rate = 1 << 22; // 4MB.
+
+    // Map stores into their URL forms.
+    let stores = stores.iter().map(|s| s.to_url().into()).collect();
+
+    // Labels must be in alphabetical order.
+    let labels = vec![
+        broker::Label {
+            name: labels::MANAGED_BY.to_string(),
+            value: labels::MANAGED_BY_FLOW.to_string(),
+        },
+        broker::Label {
+            name: labels::CONTENT_TYPE.to_string(),
+            value: labels::CONTENT_TYPE_JSON_LINES.to_string(),
+        },
+        broker::Label {
+            name: labels::COLLECTION.to_string(),
+            value: collection.to_string(),
+        },
+    ];
+
+    broker::JournalSpec {
+        name: collection.to_string(),
+        replication,
+        fragment: Some(broker::journal_spec::Fragment {
+            compression_codec,
+            flush_interval,
+            length,
+            path_postfix_template,
+            refresh_interval,
+            retention,
+            stores,
+        }),
+        flags,
+        labels: Some(broker::LabelSet { labels }),
+        max_append_rate,
+    }
+}
+
+// recovery_log_template returns a template JournalSpec for creating
+// or updating recovery logs of task shards.
+pub fn recovery_log_template(
+    task_name: &str,
+    task_type: &str,
+    stores: &[names::Store],
+) -> broker::JournalSpec {
+    use names::CompressionCodec;
+
+    // Until there's a good reason otherwise, we hard-code that recovery logs are replicated 3x.
+    let replication = 3;
+
+    // Use Snappy compression. Note that lower levels of an LSM tree
+    // typically apply their own compression, but the rocks WAL is
+    // uncompressed. Snappy has good support for passing-through content
+    // that's already compressed.
+    // TODO(johnny): Switch gazette to https://github.com/klauspost/compress/tree/master/s2
+    let compression_codec = CompressionCodec::Snappy.into_proto();
```

Comment:
We actually [do use SNAPPY today](https://github.com/estuary/flow/blob/master/go/flow/journals.go#L127) for recovery logs.

It's not a problem to change the compression of a journal at any time, FYI, because each fragment stands alone with compression identified by its file suffix.